src/Extension/FluentVersionedExtension.php

```diff
@@ -304,21 +327,26 @@ protected function isLocalisedInStage($stage, $locale = null)
             $table .= self::SUFFIX_LIVE;
         }
 
+        if (isset(self::$idsInLocaleCache[$locale][$table])) {
+            return in_array($this->owner->ID, self::$idsInLocaleCache[$locale][$table]);
```

Comment:
Checking if an array key exists is a lot faster than checking if a value is present in array. Might be worthwhile restructuring `$idsInLocaleCache` to do that instead.

Reply:
Good call! I'll change this